amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/RetrievalConfig.java

```diff
@@ -149,9 +150,9 @@ public RetrievalConfig(@NonNull KinesisAsyncClient kinesisAsyncClient, @NonNull
      */
     @Deprecated
     public RetrievalConfig initialPositionInStreamExtended(InitialPositionInStreamExtended initialPositionInStreamExtended) {
-        if (streamTracker().isMultiStream()) {
+        if (streamTracker().isMultiStream() || StreamProcessingMode.SINGLE_STREAM_MODE != streamTracker().streamProcessingMode()) {
```

Comment:
You maybe right, but I need to double check this part. But I thought, they can still use this during compatible and upgrade modes because those are also single stream mode? 

Also if you really want to do this, can just be 
```
if (streamTracker.isStreamProcessingMode() != SINGLE_STREAM_MODE) {
  ....Cannot set... when not in single stream mode...
}
```